core/src/main/java/org/mapfish/print/processor/jasper/TableProcessor.java

```diff
@@ -291,7 +291,7 @@ public Output execute(final Input values, final ExecutionContext context) throws
     // check if there are columns with mixed int and bigdecimal values
     HashSet<String> toCorrect = new HashSet<>();
     for (Map.Entry<String, Class<?>> entry : columns.entrySet()) {
-      if (entry.getValue().equals(Integer.class)) {
+      if (entry.getValue() != null && entry.getValue().equals(Integer.class)) {
```

Comment:
How about Integer.class.equals(entry.getValue()) making this null check redundant. 
Is entry != null not required ?

Reply:
Not explicitly said in the doc of `entrySet` https://docs.oracle.com/en/java/javase/17/docs/api/jdk.httpserver/com/sun/net/httpserver/Headers.html#entrySet() but for me, it's absurd that entry is `null`...